react/src/lib/components/DeckGLMap/components/Map.tsx

```diff
@@ -427,10 +430,13 @@ const Map: React.FC<MapProps> = ({
     // calculate view state on deckgl context load (based on viewport size)
     const onLoad = useCallback(() => {
         let tempViewStates: Record<string, ViewStateType> = {};
-        if (cameraPosition && Object.keys(cameraPosition).length !== 0) {
+        const isCameraPositionDefined =
```

Comment:
feels like here has a lot of duplication code have been introduced(looks exactly as from line 399). Please consider to resolve this.